### PR TITLE
[SAIC-536] CF scenarios: migration in stages.

### PR DESCRIPTION
**Pull request by archyufa**
Add CF scenarios for resource migration in stages.

Add CF scenarios for `Cold_migrate` and `Migrate_resources` with
autorollback for testing purposes.

Fix rollback steps for Migrate_resources scenario.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Pylint check succeeded

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1449/

File: scenario/stages/5_network_transporter.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Stage to migrate network resources such as networks, subnets,
+ports, router, FIPs, sec groups.
```
Comment (roman-verchikov) [inline scenario/stages/5_network_transporter.yaml]: Missing comment

File: scenario/cold_migrate_autorollback.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          - act_check_compute_api_dst: True
+          - act_check_storage_api_dst: True
+          - act_check_neutron_api_dst: True
+          - check_dst_ssh_access: True
+          - act_check_bandwidth_dst: True
+          - check_dst_sql: True
+          - check_dst_rabbit: True
+          - act_get_info_objects_dst: False
+      - check_networks: True
+  - check_users_availability: False
```
Comment (miarmak) [inline scenario/cold_migrate_autorollback.yaml]: Do we need this task here? If yes, let us set it in `True`, if no, let us remove it from here.

File: scenario/migrate_resources_autorollback.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace:
+  info_result:
+      instances: {}
+
+preparation:
+  - pre_migration_test:
+      - act_get_filter: True
+      - act_check_filter: True
+      - src_test:
```
Comment (miarmak) [inline scenario/migrate_resources_autorollback.yaml]: `check_config_quota_neutron` task is missed. Also, check if this scenario is fully up to date.

File: scenario/migrate_resources_autorollback.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          - act_check_compute_api_dst: True
+          - act_check_storage_api_dst: True
+          - act_check_neutron_api_dst: True
+          - check_dst_ssh_access: True
+          - act_check_bandwidth_dst: True
+          - check_dst_sql: True
+          - check_dst_rabbit: True
+          - act_get_info_objects_dst: False
+      - check_networks: True
+  - create_snapshot: True
```
Comment (miarmak) [inline scenario/migrate_resources_autorollback.yaml]: The same as above about `check_users_availability` task.

File: scenario/migrate_resources_autorollback.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  - create_image_snapshot_dst: True
+
+rollback:
+  - image_rollback_dst: True
+  - restore_from_snapshot: True
+
+
```
Comment (miarmak) [inline scenario/migrate_resources_autorollback.yaml]: We do not need extra space here.

File: scenario/migrate_resources_autorollback.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      - act_identity_trans: True
+      - task_images_trans:
+          - act_get_info_images: True
+          - act_deploy_images: True
+      - act_comp_res_trans: True
+      - act_network_trans: True
+      - get_volumes_db_data: True
+      - write_volumes_db_data: True
+      - transport_key_pairs: True
+      - image_rollback_dst: True
```
Comment (miarmak) [inline scenario/migrate_resources_autorollback.yaml]: Let us separate rollback tasks from the migration process.

File: scenario/stages/0_prechecks.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# Initial checks only, to ensure CF is setup correctly prior to migration.
+# This scenario checks includes: SSH connectivity to compute nodes,
+# DB access, RabbitMQ connectivity, required bandwidth verification,
+# Opentack API functionality on source  and destination clouds.
+# As well as network verification such as segmentation IDs, subnets and
+# floating IP overlap checks. And no VMs spawned directly in external network.
```
Comment (miarmak) [inline scenario/stages/0_prechecks.yaml]: Also, CF filter check ;)

File: scenario/stages/0_prechecks.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      - dst_test:
+          - act_check_image_api_dst: True
+          - act_check_compute_api_dst: True
+          - act_check_storage_api_dst: True
+          - act_check_neutron_api_dst: True
+          - check_dst_ssh_access: True
+          - act_check_bandwidth_dst: True
+          - check_dst_sql: True
+          - check_dst_rabbit: True
+      - check_networks: True
```
Comment (miarmak) [inline scenario/stages/0_prechecks.yaml]: The same as above about `check_users_availability` task.

File: scenario/stages/3_get_and_transfer_images.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Stage to migrate glance resources such as images and image members
```
Comment (miarmak) [inline scenario/stages/3_get_and_transfer_images.yaml]: And snapshots ;)

File: scenario/stages/5_network_transporter.yaml (hunk 1 above)
Comment (miarmak) [inline scenario/stages/5_network_transporter.yaml]: + Neutron quotas, LoadBalancer resources (pools, monitors, members, vips)
- ports (not all; f.e. for VMs we create them in `prepare_networks` task)

**Comment by MirantisWorkloadMobility** [comment]
Ready

**Comment by roman-verchikov** [comment]
@archyufa please resolve @miarmak's comments

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: scenario/cold_migrate_autorollback.yaml (hunk 2 above)
Comment (miarmak) [inline scenario/cold_migrate_autorollback.yaml]: So, wy don't we change it to `True?`

File: scenario/migrate_resources_autorollback.yaml (hunk 4 above)
Comment (miarmak) [inline scenario/migrate_resources_autorollback.yaml]: The same as above

File: scenario/stages/0_prechecks.yaml (hunk 8 above)
Comment (miarmak) [inline scenario/stages/0_prechecks.yaml]: The same as above

**Comment by archyufa** [comment]
Tested in Staging so far so good +1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: scenario/prechecks.yaml [hunk 10]
```diff
@@ -28,3 +28,4 @@ preparation:
           - check_dst_sql: True
           - check_dst_rabbit: True
       - check_networks: True
+  - check_users_availability: True
```
Comment (miarmak) [inline scenario/prechecks.yaml]: It seems we have 2 the same scenarios: this one and `scenario/stages/0_prechecks.yaml`. Maybe we'll stay only with one?

**Comment by miarmak** [comment]
Looks good overall. There is only 1 comment inline.

File: scenario/prechecks.yaml (hunk 10 above)
Comment (archyufa) [inline scenario/prechecks.yaml]: Is it possible to keep them ? As one used for stage resource migration and another just can be used for testing CF. It is easier from document prospective.

Comment (archyufa) [inline scenario/prechecks.yaml]: Done

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by miarmak** [comment]
Nice, thank you! 
+1 from my side.

**Comment by archyufa** [comment]
Tested in staging. +1

**Comment by roman-verchikov** [comment]
can be merged.

**Pull request merged**